compiler/ballerina-lang/src/main/java/io/ballerina/projects/BalaWriter.java

```diff
@@ -245,51 +247,72 @@ private void addPackageDoc(ZipOutputStream balaOutputStream, Path packageSourceD
             throws IOException {
         final String packageMdFileName = "Package.md";
         final String moduleMdFileName = "Module.md";
+        final String readmeMdFileName = "README.md";
 
-        Path packageMd = packageSourceDir.resolve(packageMdFileName);
         Path docsDirInBala = Paths.get(BALA_DOCS_DIR);
-
-        // If `Package.md` exists, create the docs directory & add `Package.md`
-        if (packageMd.toFile().exists()) {
-            Path packageMdInBala = docsDirInBala.resolve(packageMdFileName);
-            putZipEntry(balaOutputStream, packageMdInBala,
-                    new FileInputStream(String.valueOf(packageMd)));
-        }
+        Path packageMd = packageSourceDir.resolve(packageMdFileName);
 
         // If `icon` mentioned in the Ballerina.toml, add it to docs directory
         String icon = this.packageContext.packageManifest().icon();
         if (icon != null && !icon.isEmpty()) {
             Path iconPath = getIconPath(icon);
             Path iconInBala = docsDirInBala.resolve(iconPath.getFileName());
-            putZipEntry(balaOutputStream, iconInBala, new FileInputStream(String.valueOf(iconPath)));
+            try (FileInputStream inputStream = new FileInputStream(String.valueOf(iconPath))) {
+                putZipEntry(balaOutputStream, iconInBala, inputStream);
+            }
         }
 
-        // If `Module.md` of default module exists, create `docs/modules` directory & add `Module.md`
-        Path defaultModuleMd = packageSourceDir.resolve(moduleMdFileName);
-        Path modulesDirInBalaDocs = docsDirInBala.resolve(MODULES_ROOT);
+        // If Package.md and Module.md does not exist, pack README.md
+        if (!packageMd.toFile().exists()) {
+            packModulesToBala(pkgName, readmeMdFileName, balaOutputStream, packageSourceDir);
+        } else {
+            // Warning if Package.md or Module.md is present
+            String warning = "WARNING: Package.md is deprecated. \n" +
```

Comment:
Let's move the warning message to the PackCommand class so that it will get printed before the compilation.

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/42689/commits/f35d714b001b61eae91fbe38f327e940bf8e0c59